pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
The new format is a strict superset of the old, so these are not mutually exclusive options.

The new format adds a level of JSON around an identical structure, and at the top-level can now also include `HttpHeaders`, which you aren't using yet.

This is clearly possible via `credentialprovider`, as I linked you a PR that does precisely this (ignoring `HttpHeaders`).

To support `HttpHeaders`, the `credentialprovider` interface will have to change anyways.